### ISSUE TITLE
Implement Macro System

### ISSUE THREAD

**Issue by spy16** [spy16/slurp#4]
- [x] Identifying macro values.
- [x] Implement macro expansion.
- [ ] Handling symbol collision.

**Comment by spy16** [comment]
@lthibault I was wondering if instead of having `Expander` as another thing, we could just shift the responsibility of Expansion to Analyzer itself? What are your thoughts? So we just have single `Analyzer` interface that is responsible for all kinds of syntax analysis.

**Comment by lthibault** [comment]
@spy16 Yeah, I don't see a strong case for keeping the two separate anymore.  Let's do it 👍
